enhancements/microshift/microshift-apiserver-certs.md

```diff
@@ -0,0 +1,312 @@
+---
+title: microshift-apiserver-certs
+authors:
+  - "@pacevedom"
+reviewers:
+  - "@fzdarsky, MicroShift architect"
+  - "@ggiguash, MicroShift contributor"
+  - "@stlaz, Security specialist"
+  - "@pmtk, MicroShift contributor"
+  - "@copejon, MicroShift contributor"
+  - "@deads2k, OpenShift architect"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - N/A
+creation-date: 2023-01-20
+last-updated: 2023-01-20
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-716
+see-also:
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# MicroShift API server certificates
+## Summary
+This enhancement proposes a strategy to manage API server certificates in all
+kinds of environments, both externally (accessible through kubeconfig) and
+internally (pods reaching the API server).
+
+MicroShift has a wide variety of different use cases when it comes to API
+server connectivity, we can find environments where the cluster needs to be
+reachable with the IP, with the hostname, by some other name having DNS, or
+a combination of them.
+
+## Motivation
+MicroShift does not cover all requirements and use cases right now. It shares
+certificate handling with OCP, which is a bit different in terms of deployment.
+
+MicroShift does not have load balancers before the API server, and is required
+to be reachable and authenticate by using the IP, hostname, a DNS name, or a
+combination of those. In order to fulfill industrial use cases, we propose
+having a single API server certificate.
+
+### User Stories
+* As a MicroShift administrator/user, I want authentication by using the
+hostname.
+* As a MicroShift administrator/user, I want authentication by a name
+registered in a DNS.
+* As a MicroShift administrator/user, I want authentication by its IP address.
+* As a MicroShift application developer, I want authentication by the service
+name, `kubernetes`.
+* As a MicroShift application developer, I want authentication by the service
+IP.
+
+### Goals
+* Allow authentication from all possible sources to API server.
+
+### Non-Goals
+N/A
+
+## Proposal
+MicroShift uses several serving certificates for internal and external
+communication. Among them we can find:
+* Kubelet
+* Route controller manager
+* Router default
+* Etcd
+* API server external
+* API server localhost
+* API server service network
+
+Each of these certificates is signed by a different CA.
+
+The ones we are focusing on are the API server serving certificates. We
```

Comment:
Since this statement is essentially "what is being proposed", it would helpful if you made it more clear about how "have a single certificate for all API server connections, whether
they are internal or external" solves the problem states in the goals section "Allowing authentication from all possible sources to API server".